3506/2/checkin.rb

```diff
@@ -0,0 +1,58 @@
+module CheckIn
+  def checkin!(*)
+    error = validate_checkin
+    if error
+      respond_with :message, text: error
+    else
+      upload_checkin_photo
+    end
+  end
+
+  def validate_checkin
+    return 'Ты не зарегистрировался.' unless user
+    return 'Ты уже в лагере.' if user.in_camp == 'true'
```

Comment:
If you set boolean at first place as value of `in_camp` key, you will not need that weird check. You `user.in_camp` will return `true` boolean